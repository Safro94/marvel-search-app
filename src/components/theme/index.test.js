import React from 'react';
import { render, wait } from '@testing-library/react';

import App from './';

jest.mock('../../helpers/theme', () => ({
  light: {},
  dark: {},
}));

jest.mock('./style', () => {
  return {
    Select: 'select',
  };
});

jest.mock('../../hooks/application', () => ({
  useApplication: jest.fn(() => ({
    theme: { light: {} },
    setApplication: jest.fn(),
  })),
}));

describe('<App /> ', () => {
  beforeEach(() => jest.clearAllMocks());

  it('It should render OK', async () => {
    // Act
    const { container } = render(<App />);
    await wait();

    // Assert
    expect(container).toMatchSnapshot();
  });
});
