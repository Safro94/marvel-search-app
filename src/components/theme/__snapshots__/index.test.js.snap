// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App />  It should render OK 1`] = `
<div>
  <select>
    <option
      value="light"
    >
      light
    </option>
    <option
      value="dark"
    >
      dark
    </option>
  </select>
</div>
`;
